modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1051,4 +1158,62 @@ private CompletableFuture<Void> ensureReplicaIsPrimary(ReplicaRequest request) {
             return CompletableFuture.completedFuture(null);
         }
     }
+
+    /**
+     * Cursor wrapper to resolve a write intent, which may be required for scan by timestamp.
+     */
+    private class BinaryRowCursorWrapper implements Cursor<BinaryRow> {
```

Comment:
Why we need it?

why not to use
```
        for (int i = 0; i < batchCount && cursor.hasNext(); i++) {
            batchRows.add(resolveWriteIntent(cursor.next()));
        }
```
Where cursor will be common PartitionTimestampCursor?

Reply:
BinaryRowCursorWrapper incapsulates a logic about missing `null` values (which might be resolved for a write intent).

Reply:
Do you mean one that might be handled as follows:
```
                BinaryRow resolvedReadResult = resolveReadResult(cursor.next());
                if (resolvedReadResult != null) {
                    batchRows.add(resolvedReadResult);
                }
```
?